source/dotnet/Library/AdaptiveCards.Rendering.Html/AdaptiveCardRenderer.cs

```diff
@@ -729,6 +699,43 @@ protected static HtmlTag TextBlockRender(AdaptiveTextBlock textBlock, AdaptiveRe
             return uiTextBlock;
         }
 
+        private static string GetFontFamily(AdaptiveFontStyle fontStyle, AdaptiveHostConfig hostConfig)
+        {
+            string fontFamilyValue = hostConfig.FontStyles.GetFontStyle(fontStyle).FontFamily;
+
+            if (string.IsNullOrEmpty(fontFamilyValue))
+            {
+                fontFamilyValue = hostConfig.FontStyles.Default.FontFamily;
+                if (string.IsNullOrEmpty(fontFamilyValue))
+                {
+                    fontFamilyValue = hostConfig.FontFamily;
+                    if (string.IsNullOrEmpty(fontFamilyValue))
+                    {
+                        fontFamilyValue = "Segoe UI";
+                    }
+                }
+            }
+
+            return fontFamilyValue;
+        }
+
+        private static int GetFontWeight(AdaptiveFontStyle fontStyle, AdaptiveTextWeight requestedWeight, AdaptiveHostConfig hostConfig)
+        {
+            return hostConfig.FontStyles.GetFontStyle(fontStyle).FontWeights.GetFontWeight(requestedWeight)
+                ?? hostConfig.FontStyles.Default.FontWeights.GetFontWeight(requestedWeight)
+                ?? hostConfig.FontWeights.GetFontWeight(requestedWeight)
+                ?? FontWeightsConfig.GetDefaultFontWeight(requestedWeight);
+        }
+
+        private static int GetFontSize(AdaptiveFontStyle fontStyle, AdaptiveTextSize requestedSize, AdaptiveHostConfig hostConfig)
+        {
+            return hostConfig.FontStyles.GetFontStyle(fontStyle).FontSizes.GetFontSize(requestedSize)
+                ?? hostConfig.FontStyles.Default.FontSizes.GetFontSize(requestedSize)
+                ?? hostConfig.FontSizes.GetFontSize(requestedSize)
+                ?? FontSizesConfig.GetDefaultFontSize(requestedSize);
+        }
+
+
         protected static HtmlTag ImageRender(AdaptiveImage image, AdaptiveRenderContext context)
```

Comment:
This logic should be moved into the host config, so that
1. we can have a consistent implementation and
2. the renderer doesn't need to know anything about property deprecation.